probe/kubernetes/cronjob.go

```diff
@@ -0,0 +1,75 @@
+package kubernetes
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	batchv1 "k8s.io/client-go/pkg/apis/batch/v1"
+	batchv2alpha1 "k8s.io/client-go/pkg/apis/batch/v2alpha1"
+
+	"github.com/weaveworks/scope/report"
+)
+
+// These constants are keys used in node metadata
+const (
+	Schedule      = "kubernetes_schedule"
+	Suspended     = "kubernetes_suspended"
+	LastScheduled = "kubernetes_last_scheduled"
+	ActiveJobs    = "kubernetes_active_jobs"
+)
+
+// CronJob represents a Kubernetes cron job
+type CronJob interface {
+	Meta
+	Selectors() ([]labels.Selector, error)
+	GetNode() report.Node
+}
+
+type cronJob struct {
+	*batchv2alpha1.CronJob
+	Meta
+	jobs []*batchv1.Job
+}
+
+// NewCronJob creates a new cron job. jobs should be all jobs, which will be filtered
+// for those matching this cron job.
+func NewCronJob(cj *batchv2alpha1.CronJob, jobs []*batchv1.Job) CronJob {
+	myJobs := []*batchv1.Job{}
+	for _, j := range jobs {
+		for _, o := range cj.Status.Active {
+			if j.UID == o.UID {
+				myJobs = append(myJobs, j)
+				break
+			}
+		}
+	}
```

Comment:
This makes the complexity of the call site - `WalkCronJobs` - O(n_jobs * n_cronjobs * avg_jobs_per_cron_job). With avg_jobs_per_cron_job = n_jobs / n_cronjobs, that ends up as O(n_jobs^2).

Stuffing the jobs into a job_id -> job map in `WalkCronJobs`, instead of a list, would reduce that to O(n_jobs).